src/client.c

```diff
@@ -1,46 +1,175 @@
 #include "client.h"
 #include <assert.h>
+#include <pthread.h>
 #include <stdlib.h>
+#include <time.h>
 
+// Diferentes niveles de bloqueo en reservas.
+static pthread_mutex_t global_mutex = PTHREAD_MUTEX_INITIALIZER;
+static pthread_mutex_t* row_mutexes;
+static pthread_mutex_t* seat_mutexes;
+
+// Inicializa Datos del cliente
 void client_data_init(client_data_t* data, movie_t* movie, int id) {
     data->movie = movie;
     data->id    = id;
 }
 
+// Inicializa el resultado de la reserva.
 void client_res_init(client_res_t* res) {
-    res->success = false; // Por defecto la reserva comienza fallando.
+    res->success = false;
 }
 
-void* client_run(client_t* client) {
-    // De momento el cliente no hace nada.
+// Puntero a función. Se asignará a la función de ejecución que le corresponda.
+void* (*client_run)(void*) = NULL;
+
+// Bloqueo Global
+void* client_global_mutex(void* arg) {
+    client_t* client = (client_t*)arg;
+    movie_t* movie   = client->data.movie;
+    int id           = client->data.id;
+    bool reserved    = false;
+
+    pthread_mutex_lock(&global_mutex); // Bloqueo global
+    int rand_row = rand() % movie->nrows;
+    int rand_col = rand() % movie->ncols;
+    reserved     = movie_reserve_seat(movie, rand_col, rand_row, id);
```

Comment:
Aquí puedes asignar directamente a `client->res.success`, no hace falta una variable intermedia. Tampoco veo necesario hacer una copia del id del cliente.
```suggestion
    client->res.success = movie_reserve_seat(movie, rand_col, rand_row, client->data.id);
```